apps/storefront/src/hooks/dom/useMyQuote.ts

```diff
@@ -256,58 +240,61 @@ const useMyQuote = ({ setOpenPage, productQuoteEnabled, role }: MutationObserver
       return;
     }
 
-    if (!isBuyPurchasable) {
+    if (!isProductPurchasable) {
       clearQuoteDom();
-      const noPurchasableQuoteAll = document.querySelectorAll(config['dom.setToNoPurchasable']);
 
-      const CustomAddToQuoteAll = noPurchasableQuoteLocationSelector
-        ? document.querySelectorAll(noPurchasableQuoteLocationSelector)
+      const nonPurchasableElements = [
+        ...document.querySelectorAll(config['dom.setToNoPurchasable']),
+      ]
+        .map((el) => el.parentElement)
+        .filter((el) => el !== null);
+
+      const customNonPurchasableElements = nonPurchasableLocationSelector
+        ? [...document.querySelectorAll<HTMLElement>(nonPurchasableLocationSelector)]
         : [];
 
-      if (!noPurchasableQuoteAll.length && !CustomAddToQuoteAll.length) return;
+      if (nonPurchasableElements.length) {
+        const elements = customNonPurchasableElements.length
+          ? customNonPurchasableElements
+          : nonPurchasableElements;
 
-      if (noPurchasableQuoteAll.length) {
-        purchasableQuote(CustomAddToQuoteAll, noPurchasableQuoteAll, false);
+        renderQuoteButton(elements, false);
       }
     } else {
       clearNoPurchasableQuoteDom();
-      const addToQuoteAll = document.querySelectorAll(config['dom.setToQuote']);
-      const CustomAddToQuoteAll = locationSelector
-        ? document.querySelectorAll(locationSelector)
+
+      const defaultElements = [...document.querySelectorAll(config['dom.setToQuote'])]
+        .map((el) => el.parentElement)
+        .filter((el) => el !== null);
+
+      const customElements = locationSelector
+        ? [...document.querySelectorAll<HTMLElement>(locationSelector)]
         : [];
 
-      if (!addToQuoteAll.length && !CustomAddToQuoteAll.length) return;
-      purchasableQuote(CustomAddToQuoteAll, addToQuoteAll, true);
-    }
+      if (!defaultElements.length && !customElements.length) return;
 
-    // eslint-disable-next-line
-    return () => {
-      const myQuoteBtn = document.querySelectorAll('.b2b-add-to-quote');
-      myQuoteBtn.forEach((item: CustomFieldItems) => {
-        item.removeEventListener('click', quoteCallBack);
-      });
-    };
+      const elements = customElements.length ? customElements : defaultElements;
+      renderQuoteButton(elements, true);
+    }
```

Comment:
I noticed the before version had a cleanup function that removed event listeners. Was that intentionally removed?

Also noticed it only cleaned up `.b2b-add-to-quote` buttons but should we also clean up `.b2b-add-to-no-purchasable-quote` listeners if we add it back?

Reply:
Unresolving for visibility, feel free to resolve it back 😅